app/data_source/dynamic_loader.py

```diff
@@ -0,0 +1,96 @@
+import ast
+import os
+import re
+from dataclasses import dataclass
+from typing import Dict
+import importlib
+
+from data_source.utils import snake_case_to_pascal_case
+
+
+@dataclass
+class ClassInfo:
+    name: str
+    file_path: str
+
+
+class DynamicLoader:
+    SOURCES_PATH = 'data_source/sources'
+
+    @staticmethod
+    def extract_classes(file_path: str):
+        with open(file_path, 'r') as f:
+            file_ast = ast.parse(f.read())
+            classes = {}
+            for node in file_ast.body:
+                if isinstance(node, ast.ClassDef):
+                    classes[node.name] = {'node': node, 'file': file_path}
+        return classes
+
+    @staticmethod
+    def get_data_source_class(data_source_name: str):
+        class_name = f"{snake_case_to_pascal_case(data_source_name)}DataSource"
+        class_file_path = DynamicLoader.find_class_file(DynamicLoader.SOURCES_PATH, class_name)
+        return DynamicLoader.get_class(class_file_path, class_name)
+
+    @staticmethod
+    def get_class(file_path: str, class_name: str):
+        module_name = file_path.replace("/", ".").replace(".py", "")
+        module = importlib.import_module(module_name)
+        try:
+            return getattr(module, class_name)
+        except AttributeError:
+            raise AttributeError(f"Class {class_name} not found in module {module},"
+                                 f"make sure you named the class correctly (it should be <Platform>DataSource)")
```

Comment:
I really prefer a decorator over this approach

i.e.
```
@data_source
class ConfluenceDataSource:
```
then just import all the modules and let each data source register itself

- [ ] discuss